src/core/config/common_option.h

```diff
@@ -17,22 +17,25 @@ class CommonOption {
     std::optional<T> const default_value_;
     typename Option<T>::NormalizeFunc const normalize_func_;
     typename Option<T>::ValueCheckFunc const value_check_func_;
+    typename Option<T>::DefaultFunc const default_func_;
 
 public:
     CommonOption(std::string_view name, std::string_view description,
                  std::optional<T> default_value = std::nullopt,
                  typename Option<T>::NormalizeFunc normalize_func = nullptr,
-                 typename Option<T>::ValueCheckFunc value_check_func = nullptr)
+                 typename Option<T>::ValueCheckFunc value_check_func = nullptr,
+                 typename Option<T>::DefaultFunc default_func_ = nullptr)
```

Comment:
I don't think it should be possible to pass to the constructor simultaneously default value and default function, because it's not obvious which takes preference when both are specified. Let's better add a separate constructor with `default_func` parameter, but without `default_value`, as in `Option`